src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1724,24 +1726,7 @@ def import_on_boot(self, job):
         if os.path.exists(ZPOOL_CACHE_FILE):
             shutil.copy(ZPOOL_CACHE_FILE, zpool_cache_saved)
 
-        # Now that pools have been imported we are ready to configure system dataset,
-        # collectd and syslogd which may depend on them.
-        try:
-            self.middleware.call_sync('etc.generate', 'system_dataset')
-        except Exception:
-            self.logger.warn('Failed to setup system dataset', exc_info=True)
-
-        self.middleware.call_sync('etc.generate', 'docker')
-
-        try:
-            self.middleware.call_sync('etc.generate', 'collectd')
-        except Exception:
-            self.logger.warn('Failed to configure collectd', exc_info=True)
-
-        try:
-            self.middleware.call_sync('etc.generate', 'syslogd')
-        except Exception:
-            self.logger.warn('Failed to configure syslogd', exc_info=True)
+        self.middleware.call_sync('etc.generate_checkpoint', 'pool_import')
```

Comment:
How are we handling order? system_dataset should always come first on this checkpoint

Reply:
It is executed first because it comes first in the groups list in etc plugin then other groups which are there. However I think I should split the GROUPS list in etc plugin to make the dependencies more clear.